src/celeritas/optical/surface/detail/InitBoundaryExecutor.hh

```diff
@@ -93,8 +93,13 @@ CELER_FUNCTION void InitBoundaryExecutor::operator()(CoreTrackView& track) const
     surface_physics
         = SurfacePhysicsView::Initializer{oriented_surface.surface,
                                           oriented_surface.orientation,
+                                          geo.normal(),
                                           pre_volume_material,
                                           post_volume_material};
+
+    CELER_ASSERT(
+        is_entering_surface(geo.dir(), surface_physics.global_normal()));
```

Comment:
Working on the "geo.normal" stuff now, I don't think we should force the geometry to flip the sign of the normal based on what direction a track is traveling/what side of the surface the particle is. Here I think we should update the normal's sign as needed at the start of the boundary crossing based on the track's physical direction, and we might also need/want to keep track of the traversal direction, rather than using the physical direction `s_physics.traversal_direction(this->geometry().dir())`.